File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0, r = heights.size() - 1;
        int max_area = 0;

        while (l < r) {
            max_area = max(max_area, (r - l) * min(heights[l], heights[r]));
            if (heights[l] > heights[r]) --r;
            else ++l;
        }

        return max_area;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut  minPriv : i32 = 2147483647;
        let mut ans : i32 = 0;
        for price in &prices {
            minPriv = std::cmp::min(minPriv, *price);
            ans = std::cmp::max(ans, *price - minPriv);
        }
        ans
    }
}
